Parking_Zone.Test/Controllers/ParkingZoneControllerTest.cs

```diff
@@ -0,0 +1,308 @@
+        Assert.IsType<ViewResult>(viewResult);
+    }
+
+    [Fact]
+    public void GivenCreateVM_WhenPostCreateIsCalled_ThenModelStateIsFalseAndReturnsViewResultVM()
+    {
+        //Arrange
+        CreateVM createVM = new();
+        _parkingZoneController.ModelState.AddModelError("Name", "Required");
+
+        //Act
+        var viewResult = _parkingZoneController.Create(createVM);
+
+        //Assert
+        Assert.NotNull(viewResult);
+        Assert.IsType<ViewResult>(viewResult);
+        Assert.False(_parkingZoneController.ModelState.IsValid);
+    }
+
+    [Fact]
+    public void GivenCreateVM_WhenPostCreateIsCalled_ThenModelStateIsTrueAndReturnsRedirectToActionResult()
+    {
+        //Arrange
+        CreateVM createVM = new();
+        _parkingZoneServiceMoq.Setup(x => x.Create(It.IsAny<ParkingZone>()));
+
+        //Act
+        var viewResult = _parkingZoneController.Create(createVM);
+
+        //Assert
+        var redirectResult = Assert.IsType<RedirectToActionResult>(viewResult);
+        Assert.Equivalent(redirectResult.ActionName, "Index");
+        Assert.True(_parkingZoneController.ModelState.IsValid);
+        _parkingZoneServiceMoq.Verify(x => x.Create(It.IsAny<ParkingZone>()), Times.Once());
+    }
+
+    #endregion
+
+    #region Edit
+
+    [Fact]
+    public void GivenId_WhenEditIsCalled_ThenReturnsNotFoundResult()
+    {
+        //Arrange
+        _parkingZoneServiceMoq.Setup(x => x.GetById(id));
+
+        //Act
+        var result = _parkingZoneController.Edit(id);
+
+        //Assert
+        Assert.IsType<NotFoundResult>(result);
+        _parkingZoneServiceMoq.Verify(x => x.GetById(id), Times.Once());
+    }
+
+    [Fact]
+    public void GivenId_WhenEditIsCalled_ThenReturnsViewResultWithVM()
+    {
+        //Arrange
+        EditVM expectedEditVM = new(_parkingZone);
+
+        _parkingZoneServiceMoq
+            .Setup(x => x.GetById(id))
+            .Returns(_parkingZone);
+
+        //Act
+        var viewResult = _parkingZoneController.Edit(id);
+
+        //Assert
+        var model = Assert.IsType<ViewResult>(viewResult).Model;
+        Assert.IsAssignableFrom<EditVM>(model);
+        Assert.Equal(JsonSerializer.Serialize(expectedEditVM), JsonSerializer.Serialize(model));
+        _parkingZoneServiceMoq.Verify(x => x.GetById(id), Times.Once());
+    }
+
+    [Fact]
+    public void GivenIdAndEditVM_WhenPostEditIsCalled_ThenReturnsNotFoundResult()
+    {
+        //Arrange
+        EditVM editVM = new();
+        ParkingZone existingParkingZone = new() { Id = 2 };
+
+        _parkingZoneServiceMoq
+            .Setup(x => x.GetById(id))
+            .Returns(existingParkingZone);
+
+        //Act
+        var result = _parkingZoneController.Edit(id, editVM);
+
+        //Assert
+        Assert.IsType<NotFoundResult>(result);
+        _parkingZoneServiceMoq.Verify(x => x.GetById(id), Times.Once());
+    }
+
+    [Theory]
+    [InlineData("Name", "Required")]
+    [InlineData("Address", "Required")]
```

Comment:
here we dont need to pass parameter because instead of it we added model validation tests.